### PR TITLE
Adds a new config option to disable ambience and white noise

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Adds a new config option for disabling of ambient and white noise sounds.

Another option is just disabling the whitenoise and ambience sound on servers without a database, but I think this is preferable.

## Why It's Good For The Game
I **love** loading into the game and getting my ears instantly blasted with sounds (read: quality of life)

## Testing

finally... sweet sweet silence...

## Changelog
Nothing player-facing.

### CODE REVIEW

File: code/controllers/subsystem/SSambience.dm [hunk 1]
```diff
@@ -10,6 +10,10 @@ SUBSYSTEM_DEF(ambience)
 	var/list/ambience_listening_clients = list()
 
 /datum/controller/subsystem/ambience/fire(resumed)
+	if(GLOB.configuration.general.disable_ambient_noise)
+		flags |= SS_NO_FIRE
+		return
+
 	for(var/C in ambience_listening_clients)
```
Comment (Contrabang) [inline code/controllers/subsystem/SSambience.dm:17]: If anyone has a better recommendation for how to handle this, I would appreciate it.

Comment (S34NW) [reply code/controllers/subsystem/SSambience.dm:17]: @AffectedArc07 sanity check for this please?

File: code/controllers/subsystem/SSambience.dm [hunk 2]
```diff
@@ -10,6 +10,10 @@ SUBSYSTEM_DEF(ambience)
 	var/list/ambience_listening_clients = list()
 
 /datum/controller/subsystem/ambience/fire(resumed)
+	if(GLOB.configuration.general.disable_ambient_noise)
```
Comment (AffectedArc07) [inline code/controllers/subsystem/SSambience.dm:13]: Do this in init please

**Comment by AffectedArc07** [comment]
Config change committed to production repo as no real change on merge

File: code/controllers/subsystem/SSambience.dm (hunk 2 above)
Comment (Contrabang) [reply code/controllers/subsystem/SSambience.dm:13]: Do the subsystems init after the configuration is loaded? That was one thing I was worried about

File: code/controllers/subsystem/SSambience.dm [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 	runlevels = RUNLEVEL_GAME | RUNLEVEL_POSTGAME
 	wait = 1 SECONDS
 	cpu_display = SS_CPUDISPLAY_LOW
 	///Assoc list of listening client - next ambience time
 	var/list/ambience_listening_clients = list()
 
+/datum/controller/subsystem/ambience/Initialize()
+	if(GLOB.configuration.general.disable_ambient_noise)
+		flags |= SS_NO_FIRE
+		return
```
Comment (AffectedArc07) [inline code/controllers/subsystem/SSambience.dm:15]: Remove return, unnecessary.

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**